peps/pep-0751.rst

```diff
@@ -0,0 +1,901 @@
+PEP: 751
+Title: A file format to list Python dependencies for installation reproducibility
+Author: Brett Cannon <brett@python.org>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 24-Jul-2024
+Replaces: 665
+
+Abstract
+========
+
+This PEP proposes a new file format for specifying what dependencies to
+(possibly) install into a Python environment for consistent installation
+repoducibility. The format is designed to be human-readable but
+machine-generated. Installers consuming the file should not need to resolve
+which listed dependencies to install based on other packages listed in the lock
+file, but instead evaluate each dependency in question in isolation.
+
+
+Motivation
+==========
+
+Currently, there is no standard way to specify what top-level dependencies one
+would like to see installed into a Python environment and then subsequently
+record what was eventually installed in a file (called a *lock file*).
+Considering there are four well-known solutions to this problem in the
+community (``pip freeze``, pip-tools_, Poetry_, and PDM_), there seems to be an
+appetite for lock files in general.
+
+Those tools also vary in what locking scenarios they support. For instance,
+``pip freeze`` and pip-tools only generate lock files for the current
+environment while PDM and Poetry try to lock for *any* environment to some
+degree. And none of them directly support locking to specific files to install
+which can be important for some workflows. There's also concerns around secure
+defaults in the face of supply chain attacks. Finally, not all the formats are
```

Comment:
This seems a bit vague to me, can we be more specific either about the concerns or what relation they have to lock files? E.g. as a made-up example, "A standardised format would allow for security audit tools to check for secure defaults"

Reply:
It may suffice to say that these tools use various approaches to create an installation record or lock file.